span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/ExcludeSpanRulesCache.java

```diff
@@ -46,7 +48,14 @@ private ExcludeSpanRulesCache(Config config) {
 
     ConfigServiceConfig configServiceConfig = new ConfigServiceConfig(config);
     ConfigServiceClient configServiceClient =
-        new ConfigServiceClient(configServiceConfig, new GrpcChannelRegistry());
```

Comment:
pre-existing, but this client should not have its own registry, it needs to have that passed in from the span normalizer. This one would not be managed or shut down when the app is shut down.